repowiz-infrastructure/src/main/groovy/life/qbic/repowiz/prepare/projectSearch/OpenBisProjectSearcher.groovy

```diff
@@ -18,37 +18,55 @@ import ch.ethz.sis.openbis.generic.dssapi.v3.IDataStoreServerApi
 import ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.DataSetFile
 import ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.fetchoptions.DataSetFileFetchOptions
 import ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.search.DataSetFileSearchCriteria
-import life.qbic.repowiz.io.JsonParser
 import life.qbic.repowiz.model.RepoWizProject
 import life.qbic.repowiz.model.RepoWizSample
 import life.qbic.repowiz.prepare.openBis.ConditionParser
 import life.qbic.xml.properties.Property
 import org.apache.logging.log4j.LogManager
 import org.apache.logging.log4j.Logger
 
-//todo separate code, try to create general class for plugin architecture and move openBis code into concrete opnebis class
+/**
+ * This class handles the project search in OpenBis
+ *
+ * In order to fetch data from a local database and transfer it into RepoWiz the {@link ProjectSearchInput} must be implemented.
+ * Furthermore, the {@link ProjectSearcher} must be implemented.
+ *
+ *  @since: 1.0.0
+ *  @author: Jennifer Bödker
+ *
+ */
 class OpenBisProjectSearcher extends ProjectSearcher implements ProjectSearchInput {
 
     private static final Logger LOG = LogManager.getLogger(OpenBisProjectSearcher.class)
 
-    Project project
-
-    IApplicationServerApi v3
-    IDataStoreServerApi dss
-    String sessionToken
-
-    OpenBisMapper mapper
-    ConditionParser conditionParser
-
-    OpenBisProjectSearcher(IApplicationServerApi v3, IDataStoreServerApi dss, String session, OpenBisMapper mapper, String projectSchema, String sampleSchema) {
+    private Project project
+
+    private final IApplicationServerApi v3
+    private final IDataStoreServerApi dss
```

Comment:
```suggestion
    private final IDataStoreServerApi dataStoreServer
```
dss -> decision support system, ...

Reply:
d1cf104, did it with the IDE to make sure all variable occurrences get updated